src/lambda_function/template.yaml

```diff
@@ -0,0 +1,86 @@
+AWSTemplateFormatVersion: '2010-09-09'
+
+Transform: AWS::Serverless-2016-10-31
+
+Description: >
+  SAM Template for s3-to-glue lambda function
+
+Parameters:
+
+  S3SourceBucketName:
+    Type: String
+    Description: Name of the S3 bucket where source data are stored.
+
+  Namespace:
+    Type: String
+    Description: Namespace of the stack so that it may be unique
+
+  Schedule:
+    Description: >
+      Schedule to execute the lambda function
+    Type: String
+    Default: cron(59 23 * * ? *)
+
+Resources:
+  S3ToGlueRole:
+      Type: AWS::IAM::Role
+      Properties:
+        AssumeRolePolicyDocument:
+          Version: '2012-10-17'
+          Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+              - lambda.amazonaws.com
+            Action:
+            - sts:AssumeRole
+        ManagedPolicyArns:
+          - arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
+        Policies:
+        - PolicyName: StartGlueWorkflow
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+            - Effect: Allow
+              Action:
+              - glue:StartWorkflowRun
+              - glue:PutWorkflowRunProperties
+              Resource:
+              - !Sub arn:aws:glue:${AWS::Region}:${AWS::AccountId}:workflow/*
+        - PolicyName: AccessS3BucketInputData
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+            - Effect: Allow
+              Action:
+               - s3:Get*
+               - s3:List*
+              Resource:
+              - !Sub arn:aws:s3:::${S3SourceBucketName}
```

Comment:
Is this also duplicated from BridgeDownstream? Will it need access to multiple buckets?

Reply:
This lambda function template is slightly different from BridgeDownstream's lambda template. It only needs access to the source data's bucket. It uses a scheduled event trigger instead of sns/sqs.